torch/__init__.pyi.in

```diff
@@ -60,6 +59,18 @@ _size = Union[Size, List[_int], Tuple[_int, ...]]
 # Meta-type for "numeric" things; matches our docs
 Number = Union[builtins.int, builtins.float]
 
+class Generator:
+    device: _device = ...
+
+    @overload
+    def __init__(self, device: _device, default: bool) -> None: ...
+
+    @overload
+    def __init__(self, device='cpu', default=False) -> None: ...
+
+    @overload
+    def __init__(self, default: bool, device='cpu') -> None: ...
```

Comment:
Um... This looks questionable. Why do you need the latter two overloads? Is this trying to control kwargs?

Reply:
I have no idea why I needed the two overloads, but mypy/typehints was complaining when it extracted the code out of the docstrings I added, and adding this combination of overloads solved it. This doesn't matter now since we removed the default argument.

Reply:
Seems like mypy is not happy with just one overload. How does this work, do you know?

Reply:
I think, if you only have "one" overload, it's not an overload and you shouldn't use `@overload`. But honestly, the manual is the best source of truth here https://mypy.readthedocs.io/en/latest/more_types.html